File: SeparatorTextField/ViewController.m
//
//  ViewController.m
//  SeparatorTextField
//
//  Created by youngliuxx on 13/12/2017.
//  Copyright © 2017 youngliuxx. All rights reserved.
//

#import "ViewController.h"
#import "LYSeparatorTextField.h"

#define SCREEN_WIDTH ([UIScreen mainScreen].bounds.size.width)

@interface ViewController ()

@property (nonatomic, strong) LYSeparatorTextField *textField;
@property (nonatomic, strong) LYSeparatorTextField *textField2;
@property (nonatomic, strong) LYSeparatorTextField *textField3;
@property (nonatomic, strong) LYSeparatorTextField *textField4;
@property (nonatomic, strong) LYSeparatorTextField *textField5;
@property (nonatomic, strong) UIButton *submitBtn;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupViews];
}

- (void)setupViews
{
    CGFloat space = 20.0;
    
    // 12345-123456
    _textField = [[LYSeparatorTextField alloc] initWithFrame:CGRectMake(space, 100, SCREEN_WIDTH - space*2, 30)
                                                   gapNumber:4
                                                   separator:@"-"];
    _textField.placeholder = @"separator: -";
    _textField.keyboardType = UIKeyboardTypeNumberPad;
    _textField.limitCount = 12;
    [self.view addSubview:_textField];
    
    // 123,234
    _textField2 = [[LYSeparatorTextField alloc] initWithFrame:CGRectMake(space, 160, SCREEN_WIDTH - space*2, 30)
                                                   gapNumber:3
                                                   separator:@","];
    _textField2.placeholder = @"separator: ,";
    [self.view addSubview:_textField2];
    
    // xxxx-xx-xx
    _textField3 = [[LYSeparatorTextField alloc] initWithFrame:CGRectMake(space, 220, SCREEN_WIDTH - space*2, 30) format:@"^^^^-^^-^^"];
    _textField3.placeholder = @"format: xxxx-xx-xx";
    _textField3.keyboardType = UIKeyboardTypeNumberPad;
    [self.view addSubview:_textField3];
    
    // Day:01 Month:01 Year:2017
    _textField4 = [[LYSeparatorTextField alloc] initWithFrame:CGRectMake(space, 280, SCREEN_WIDTH - space*2, 30) format:@"Day:^^ Month:^^ Year:^^^^"];
    _textField4.placeholder = @"format: Day:^^ Month:^^ Year:^^^^";
    _textField4.keyboardType = UIKeyboardTypeNumberPad;
    [self.view addSubview:_textField4];
    
    // Day:01 Month:01 Year:2017
    _textField5 = [[LYSeparatorTextField alloc] initWithFrame:CGRectMake(space, 340, SCREEN_WIDTH - space*2, 30) format:@"^^^-^^^^-^^^^"];
    _textField5.placeholder = @"format: 185-0000-0001";
    _textField5.keyboardType = UIKeyboardTypeNumberPad;
    [self.view addSubview:_textField5];
    
    _submitBtn = [[UIButton alloc] initWithFrame:CGRectMake(space, 560, SCREEN_WIDTH - space*2, 40)];
    [_submitBtn setBackgroundColor:[UIColor grayColor]];
    [_submitBtn setTitle:@"Print Input Text" forState:UIControlStateNormal];
    [self.view addSubview:_submitBtn];
    [_submitBtn addTarget:self action:@selector(tapBtn:) forControlEvents:UIControlEventTouchUpInside];
    
    UITapGestureRecognizer *tapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(keyboardHide:)];
    tapGestureRecognizer.cancelsTouchesInView = NO;
    [self.view addGestureRecognizer:tapGestureRecognizer];
}

-(void)keyboardHide:(UITapGestureRecognizer *)tap{
    [_textField resignFirstResponder];
    [_textField2 resignFirstResponder];
    [_textField3 resignFirstResponder];
    [_textField4 resignFirstResponder];
    [_textField5 resignFirstResponder];
}

- (void)tapBtn:(id)sender
{
    NSLog(@"Input text:%@", _textField.getInputText);
    NSLog(@"Input text:%@", _textField2.getInputText);
    NSLog(@"Input text:%@", _textField3.getInputText);
    NSLog(@"Input text:%@", _textField4.getInputText);
    NSLog(@"Input text:%@", _textField5.getInputText);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: SeparatorTextField_Swift/ViewController.swift
//
//  ViewController.swift
//  SeparatorTextField_Swift
//
//  Created by youngliuxx on 20/12/2017.
//  Copyright © 2017 youngliuxx. All rights reserved.
//

import UIKit
import LYSeparatorTextField
let space = CGFloat(20.0)
let SCREEN_WIDTH = UIScreen.main.bounds.size.width

class ViewController: UIViewController {
    // 12345-123456
    let textField: LYSeparatorTextField = {
        () -> LYSeparatorTextField in
        let temp = LYSeparatorTextField.init(frame: CGRect.init(x: space, y: 100, width: SCREEN_WIDTH - space*2, height: 30), gapNumber: 4, separator: "-")
        temp.placeholder = "separator: -"
        temp.keyboardType = .numberPad
        temp.limitCount = 12
        return temp
    }()
    
    // 123,234
    let textField2: LYSeparatorTextField = {
        () -> LYSeparatorTextField in
        let temp = LYSeparatorTextField.init(frame: CGRect.init(x: space, y: 160, width: SCREEN_WIDTH - space*2, height: 30), gapNumber: 3, separator: ",")
        temp.placeholder = "separator: ,"
        return temp
    }()
    
    // 2017-12-15
    let textField3: LYSeparatorTextField = {
        () -> LYSeparatorTextField in
        let temp = LYSeparatorTextField.init(frame: CGRect.init(x: space, y: 220, width: SCREEN_WIDTH - space*2, height: 30), formatStr: "^^^^-^^-^^")
        temp.placeholder = "format: 2017-12-15"
        temp.keyboardType = .numberPad
        return temp
    }()
    
    // Day:01 Month:01 Year:2017
    let textField4: LYSeparatorTextField = {
        () -> LYSeparatorTextField in
        let temp = LYSeparatorTextField.init(frame: CGRect.init(x: space, y: 280, width: SCREEN_WIDTH - space*2, height: 30), formatStr: "Day:^^ Month:^^ Year:^^^^")
        temp.placeholder = "format: Day:05 Month:12 Year:2017"
        temp.keyboardType = .numberPad
        return temp
    }()
    
    // format: 185-0000-0001
    let textField5: LYSeparatorTextField = {
        () -> LYSeparatorTextField in
        let temp = LYSeparatorTextField.init(frame: CGRect.init(x: space, y: 340, width: SCREEN_WIDTH - space*2, height: 30), formatStr: "^^^-^^^^-^^^^")
        temp.placeholder = "format: 185-0000-0001"
        temp.keyboardType = .numberPad
        return temp
    }()
    
    let submitBtn: UIButton = {
        () -> UIButton in
        let temp = UIButton.init(frame: CGRect.init(x: space, y: 460, width: SCREEN_WIDTH - space*2, height: 30))
        temp.backgroundColor = UIColor.gray
        temp.setTitle("Print Input Text", for: .normal)
        temp.addTarget(self, action: #selector(tapBtn), for: .touchUpInside)
        return temp
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.addSubview(textField)
        self.view.addSubview(textField2)
        self.view.addSubview(textField3)
        self.view.addSubview(textField4)
        self.view.addSubview(textField5)
        self.view.addSubview(submitBtn)
        
        let tapGes = UITapGestureRecognizer.init(target: self, action: #selector(keyboardHide))
        self.view.addGestureRecognizer(tapGes)
    }
    
    @objc func tapBtn() {
        print("Input text:\(textField.getInputText())")
        print("Input text:\(textField2.getInputText())")
        print("Input text:\(textField3.getInputText())")
        print("Input text:\(textField4.getInputText())")
        print("Input text:\(textField5.getInputText())")
    }
    
    @objc func keyboardHide() {
        textField.resignFirstResponder()
        textField2.resignFirstResponder()
        textField3.resignFirstResponder()
        textField4.resignFirstResponder()
        textField5.resignFirstResponder()
    }
}

